zha/application/platforms/select.py

```diff
@@ -75,7 +76,10 @@ def __init__(
         """Init this select entity."""
         self._cluster_handler: ClusterHandler = cluster_handlers[0]
         self._attribute_name = self._enum.__name__
-        self._attr_options = [entry.name.replace("_", " ") for entry in self._enum]
+        self._translation_keys = {
+            entry.name.lower(): entry.name for entry in self._enum
+        }
+        self._attr_options = [entry.name.lower() for entry in self._enum]
```

Comment:
Will this be a breaking change for entities that do not yet implement translation keys?

Reply:
In homeassistant/core, I have written translation keys for all enums. But unfortunately, the internal state will change from the current form (snake_case with the first letter, and sometimes multiple letters, being a capital) to snake case without any capitals. Automations that depend on this will seize to function.

Reply:
Hmm. If it's going to be a breaking change we should get these strings into the exact state we want them (with no compatibility shims) and then prominently mark the breaking change as such.

Reply:
Currently, the only thing we do is .lower() as most Enums have capitals in them. Do you see that as a compatibility shim? Changing that requires changes in at least: zigpy/zha, zigpy/zigpy and zigpy/zha-device-handlers. There might be more users of zigpy/zigpy that rely on the names in the Enums.

Reply:
Secondly, if we want to create final names, we probably should change the CamelCase Enums, which aren't many, but they do exist in zigpy/zha and zigpy/zha-device-handlers. Thirdly, maybe this is also a nice opportunity to move the Enums still present in zigpy/zha to zigpy/zha-device-handlers, while changing the naming where appropriate.

Reply:
I've spoken to a few Core members and they said explained that unfortunately there is no migration path, this must be a breaking change.

The good news is that I looked over the [affected entities](https://github.com/home-assistant/core/pull/109309/files) once more and truthfully, they don't seem like something someone is going to automate with, making this a breaking change that would affect very very few people.

@TheJulianJES @dmulcahey thoughts?

Reply:
Yeah, seems good.

Reply:
To make this a one-time breaking change, I wonder if we can add a unit test to validate all enums in quirks to make sure they conform to the expected format?

Reply:
I mean, yeah, but isn't the expected format just anything (ASCII characters) and underscores?
Like, we just lower-case all enum values here. They won't contain spaces, so..

Reply:
`SomeValue` would become `somevalue` instead of `some_value`. I think in zigpy we use a mix of `UPPERCASE_WITH_UNDERSCORES`, `CamelCase`, and `Whatever_case_this_is`.

Reply:
Ok, I see. They're all still valid at least 😄 
So, for HA state/translation keys, we probably always want `some_value`.

Which style do we want to use for the actual enums in general? Also `some_value`?
Or do we want to leave them as-is for now and just convert all types you mentioned to `some_value` in ZHA? Should be possible at least.. (We could change the actual enums to `some_value` (or similar) later, without breaking HA state again then.)

Reply:
When leaving the actual enums as-is, from "SomeValue" we would get "somevalue" using ``.lower()``. So when changing the actual enums later on to "some_value", we would still have a breaking change. Although we could also do something like: ``re.sub(r"([A-Z])", r"_\1", "SomeValue").strip("_").lower()``.

Reply:
Here's all of the enum members defined in quirks and zigpy: https://gist.github.com/puddly/2190cc8450bbcb6200afd6b87e516252.

I think the function in there should work, I didn't see any mistakes for this mapping:

```python
{
    "FooBar": "foo_bar",
    "FOOBAR": "foobar",
    "FOO_BAR": "foo_bar",
    "foo_bar": "foo_bar",
    "FOOBar": "foo_bar"
}
```

Reply:
Yeah, let's use this.